src/Framework/AspNetCoreAnalyzers/test/Mvc/AuthorizeAttributeOverriddenTest.cs

```diff
@@ -0,0 +1,602 @@
+            new DiagnosticResult(DiagnosticDescriptors.AuthorizeAttributeOverridden).WithArguments("MyControllerBaseBase").WithLocation(0)
+        );
+    }
+
+    [Fact]
+    public async Task CustomAuthorizeOnAction_CustomAllowAnonymousOnController_HasDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+[MyAllowAnonymous]
+public class MyController
+{
+    [{|#0:MyAuthorize|}]
+    public object Get() => new();
+}
+
+public class MyAuthorizeAttribute : Attribute, IAuthorizeData
+{
+    public string? Policy { get; set; }
+    public string? Roles { get; set; }
+    public string? AuthenticationSchemes { get; set; }
+}
+
+public class MyAllowAnonymousAttribute : Attribute, IAllowAnonymous
+{
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source,
+            new DiagnosticResult(DiagnosticDescriptors.AuthorizeAttributeOverridden).WithArguments("MyController").WithLocation(0)
+        );
+    }
+
+    [Fact]
+    public async Task AuthorizeOnAction_NonInheritableAllowAnonymousOnController_NoDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+[MyAllowAnonymous]
+public class MyControllerBase
+{
+}
+
+public class MyController : MyControllerBase
+{
+    [Authorize]
+    public object Get() => new();
+}
+
+[AttributeUsage(AttributeTargets.Class, Inherited = false)]
+public class MyAllowAnonymousAttribute : Attribute, IAllowAnonymous
+{
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source);
+    }
+
+    [Fact]
+    public async Task CustomAuthorizeCombinedWithAllowAnonymousOnAction_AllowAnonymousOnController_NoDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+[AllowAnonymous]
+public class MyController
+{
+    [MyAuthorize]
+    public object Get() => new();
+}
+
+public class MyAuthorizeAttribute : Attribute, IAuthorizeData, IAllowAnonymous
+{
+    public string? Policy { get; set; }
+    public string? Roles { get; set; }
+    public string? AuthenticationSchemes { get; set; }
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source);
+    }
+
+    [Fact]
+    public async Task AuthorizeBeforeAllowAnonymousOnAction_AllowAnonymousOnController_NoDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+[AllowAnonymous]
+public class MyController
+{
+    [Authorize(AuthenticationSchemes = "foo")]
+    [AllowAnonymous]
+    public object Get() => new();
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source);
+    }
+
+    [Fact]
+    public async Task AuthorizeAfterAllowAnonymousOnAction_NoAttributeOnController_HasDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+public class MyController
+{
+    [AllowAnonymous]
+    [{|#0:Authorize|}]
+    public object Get() => new();
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source,
+            new DiagnosticResult(DiagnosticDescriptors.AuthorizeAttributeOverridden).WithArguments("MyController.Get").WithLocation(0)
+        );
+    }
```

Comment:
Should we really warn in this case? I understand that when `AllowAnonymous` is on a controller its problematic, but here it should be obvious that the action is not `secure` in the sense that it doesn't enforce authorization.

Reply:
An example of this (and I might be biased since I have experience in this field and I can probably agree it's not the common case) is for example when you are implementing multi-factor auth or social logins, where you have an action that receives the callback and has a custom `Authorize(Scheme = "Social")` to make sure the cookie is read and the principal set on the request, and that checks within the action if the principal has the right information on it or performs a custom action if not.